projects/cm_mcu/LocalTasks.c

```diff
@@ -656,3 +656,136 @@ void init_registers_ff()
+      status = apollo_i2c_ctl_reg_w(2, i2c_addrs, 1, (uint16_t)reg1, 1, data); //write data to a clock chip
+
+
+      if (status != 0){
+        log_error(LOG_SERVICE, "write status is %d \r\n",status);
+        return status; // fail writing and exit
+      }
+
+
+
+    }
+
+  }
+  return status;
+}
+
+int init_load_clk(int clk_n)
+{
+
+  while (getPowerControlState() != POWER_ON) {
+    vTaskDelay(pdMS_TO_TICKS(3000)); // 300 ms minimum
+  }
+
+
+  int status = -10;
+  char *clk_ids[5] = {"r0a","r0b","r1a","r1b","r1c"};
+  uint8_t i2c_addrs = 0x6b; // i2c address of a clock chip
+  if (clk_n == 0)
+    i2c_addrs = 0x77;
+
+  apollo_i2c_ctl_w(2, 0x70, 1, 1<<clk_n);
+  char string[5]; //place holder for a string to be converted from base16 to base10
+  sprintf(string, "%x",32*(clk_n));
+  uint16_t init_preamble_page = strtoul(string, NULL, 16);
+  sprintf(string, "%x",32*(clk_n)+1);
+  uint16_t init_register_page = strtoul(string, NULL, 16);
+  sprintf(string, "%x",32*(clk_n + 1)-1);
+  uint16_t init_postamble_page = strtoul(string, NULL, 16);
+
+  uint32_t PreambleList_row; //the hexadecimal size of preamble list in a clock config file store at the end of the last eeprom page of a clock
+  apollo_i2c_ctl_reg_r(2, 0x50, 2, (init_postamble_page << 8) + 0x007C, 1, &PreambleList_row);
+  sprintf(string, "%lu",PreambleList_row);
+  int PreambleList_nrow = strtoul(string, NULL, 10); //the decimal size of preamble list in a clock config file
+  uint32_t RegisterList_row; //the hexadecimal size of register list in a clock config file store at the end of the last eeprom page of a clock
+  apollo_i2c_ctl_reg_r(2, 0x50, 2, (init_postamble_page << 8) + 0x007D, 2, &RegisterList_row);
+  sprintf(string, "%lu",RegisterList_row);
+  int RegisterList_nrow = strtoul(string, NULL, 10); //the decimal size of register list in a clock config file
+  uint32_t PostambleList_row; //the hexadecimal size of postamble list in a clock config file store at the end of the last eeprom page of a clock
+  apollo_i2c_ctl_reg_r(2, 0x50, 2, (init_postamble_page << 8) + 0x007F, 1, &PostambleList_row);
+  sprintf(string, "%lu",PostambleList_row);
+  int PostambleList_nrow = strtoul(string, NULL, 10); //the decimal size of postamble list in a clock config file
+
+
+  log_debug(LOG_SERVICE, "Start programming clock %s \r\n", clk_ids[clk_n]);
+  log_info(LOG_SERVICE, "Loading clock %s PreambleList from EEPROM \r\n", clk_ids[clk_n]);
+  status = load_register(PreambleList_nrow, init_preamble_page, i2c_addrs);
+  if (status != 0)
+    return status;
+  vTaskDelay(pdMS_TO_TICKS(3000)); //300 ms minimum
+  log_info(LOG_SERVICE, "Loading clock %s RegisterList from EEPROM \r\n", clk_ids[clk_n]);
+  status = load_register(RegisterList_nrow, init_register_page, i2c_addrs);
+  if (status != 0)
+    return status;
+  vTaskDelay(pdMS_TO_TICKS(3000)); //300 ms minimum
+  log_info(LOG_SERVICE, "Loading clock %s PostambleList from EEPROM \r\n", clk_ids[clk_n]);
```

Comment:
these two messages are probably more relevant for the `debug` level (ie usually not interesting)

Reply:
okay.